### ISSUE TITLE
Low line heights with long text causes vertical spacing problems on smaller screens

### ISSUE THREAD

**Issue by inaikem** [WordPress/gutenberg#33536]
### Steps to reproduce the behavior
- Create a cover block and put 3 paragraph blocks inside it. 
- Set their line-height to 0.2
- Enter long lines of text into each block, so they will wrap around on a smaller screen.
- Preview the page.

### What I expected to happen
- Text should wrap and multiple lines should not overlap.

### What actually happened
- Text wraped and multiple lines collapsed vertically in on themselves.

### Context
- Chrome, Safari and Firefox / macOS

### Is this specific to the applied theme? Which one?
- Replicable on all themes.

### Screenshots
**Desktop View:**
&nbsp;
![desktop-line-height](https://user-images.githubusercontent.com/66711744/126138472-644f1b64-f50c-4fb8-9fb6-b0b1fd354760.png)

&nbsp;
**Mobile View:**
![mobile-line-height](https://user-images.githubusercontent.com/66711744/126138496-77defee6-58ac-453f-80ad-5a1d76b1973b.png)

&nbsp;

### Suggestions
I believe this is more a training/education issue than a platform bug.

We should allow users to set whatever line-height they want, as there may be a design reason for doing so. However, we could consider adding a warning if line heights are set to <1 per the below.

We provide a warning for low-contrast colour combinations and could reuse this method to notify users of potential issues.
&nbsp;

**Existing low contrast text warning:**
&nbsp;
<img width="1176" alt="Screen Shot 2021-07-19 at 11 27 01" src="https://user-images.githubusercontent.com/66711744/126139393-8d8a1b6d-6954-4572-a496-5265624e3913.png">

&nbsp;
**Suggested low line height warning 1:**
&nbsp;
![Screen Shot on 2021-07-19 at 11_33_56](https://user-images.githubusercontent.com/66711744/126139328-51d3daff-1916-4edc-b106-5ac2e939bd06.png)

&nbsp;
**Suggested low line height warning 2:**
&nbsp;
![line-height-warning-v4](https://user-images.githubusercontent.com/66711744/126139357-25da28ff-c787-40fc-ab67-6da0e95aab94.png)

cc: @mrfoxtalbot, @KirkwallDay

**Comment by richtabor** [comment]
> We should allow users to set whatever line-height they want, as there may be a design reason for doing so. 

This is why I don't think we should have a warning. I wouldn't want to hamper creative expression.
